File: InfPrecisionDouble/2/InfPrecFloat.py

class InfPrecFloat:
    def __init__(self, num):

        self.vorzeichen = num[0] != '-'
        if not self.vorzeichen:
            num = num[1:]

        self.vor_komma = []
        self.nach_komma = []
        vor, nach = num.split('.')
        self.set_vor_komma(vor)
        self.set_nach_komma(nach)

    
    def set_vor_komma(self, vor):
        for c in vor[::-1]:
            self.vor_komma.append(int(c))
    
    def set_nach_komma(self, nach):
        for c in nach:
            self.nach_komma.append(int(c))
    
    def add_vor_komma(self, sinn, vor):
        new_vor = []

        while len(self.vor_komma) < len(vor):
            self.vor_komma.append(0)
        while len(vor) < len(self.vor_komma):
            vor.append(0)
        
        for s_el, o_el in zip(self.vor_komma, vor):
            temp = sinn + s_el + o_el
            sinn = temp // 10
            new_vor.append(temp % 10)
        
        if sinn > 0:
            new_vor.append(sinn)

        return new_vor

    def add_nach_komma(self, nach):
        new_nach = []
        while len(self.nach_komma) < len(nach):
            self.nach_komma.append(0)
        
        while len(nach) < len(self.nach_komma):
            nach.append(0)

        sinn = 0
        for s_el, o_el in zip(self.nach_komma[::-1], nach[::-1]):
            temp = sinn + s_el + o_el
            sinn = temp // 10
            new_nach.append(temp % 10)

        return sinn, new_nach[::-1]



    
    def __add__(self, other):
        if self.vorzeichen and not other.vorzeichen:
            temp = other
            temp.vorzeichen = True
            return self - temp
        sinn, new_nach = self.add_nach_komma(other.nach_komma)
        new_vor = self.add_vor_komma(sinn, other.vor_komma)

        temp = InfPrecFloat('0.0')
        temp.vor_komma = new_vor
        temp.nach_komma = new_nach

        return temp
    
    def __sub__(self, other):
        print('Subtraktion')


    def __str__(self):
        out = '' if self.vorzeichen else '-'
        for el in self.vor_komma[::-1]:
            out += str(el)
        out += '.'
        for el in self.nach_komma:
            out += str(el)
        
        return out



if __name__ == '__main__':
    myVar = InfPrecFloat('3.241')
    myVar += InfPrecFloat('1.9111111111111111111111111111111111111111111111111111111155555')
    print(myVar)

File: InfPrecisionDouble/1/InfPrecFloat.cpp
#include "InfPrecFloat.h"

vector<string> split(string str, char c)
{
    vector<string> end;
    string temp = "";

    for ( auto&& i : str)
    {
        if ( i != c )
            temp += i;
        else
        {
            end.push_back(temp);
            temp = "";
        }
        
    }
    end.push_back(temp);

    return end;
}

InfPrecFloat::InfPrecFloat(string str)
{
    this->add(str);
}

void InfPrecFloat::add(string x)
{
    vector<string> temp = split(x, '.'); 
    this->add_vor_komma(temp[0]);
    this->add_nach_komma(temp[1]);
    
}

void InfPrecFloat::add_nach_komma(string x)
{
    cout << "1\n";
    for ( int i = 0; i < x.length() - nach_komma.size(); i++ )
    {
        nach_komma.push_back(0);
    }
    cout << "2\n";
    cout << nach_komma.size() << " " << x.length() << endl;
    for ( int i = 0; i < nach_komma.size() - x.length(); i++ )
    {
        x += "0";
    }
    cout << "3\n";
    int temp;
    int sinn = 0;
    for ( int i = x.length() - 1; i >= 0; i-- )
    {
        cout << i << endl;
        temp = (int) x[i] + nach_komma[i] + sinn;
        sinn = floor(temp/10);
        nach_komma[i] = temp % 10;
    }

    this->add_vor_komma(to_string(sinn));
}

void InfPrecFloat::add_vor_komma(string x)
{
    cout << "Adde Vorkomma " << x << endl;
}

InfPrecFloat::~InfPrecFloat(){}

std::ostream& operator << (std::ostream &output, InfPrecFloat &m){
    for ( int i = m.vor_komma.size() - 1; i >= 0; i-- )
        output << m.vor_komma[i];
    output << ".";
    for ( auto i : m.nach_komma )
        output << i;
    return output;
}